### PR TITLE
[2108][7538864]control remidiation undo fix

### PR DESCRIPTION
**Pull request by anandsinghms**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [x] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [x] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [x] The title of the PR clearly describes the intent of the PR.
- [x] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 1]
```diff
@@ -416,19 +416,23 @@ class AgentPool: ADOSVTBase
...
                     # Checking whether the broader groups have User/Admin permissions
                     $restrictedGroups = @($roleAssignments | Where-Object { $restrictedBroaderGroupsForAgentPool -contains $_.Name.split('\')[-1] -and ($_.Role -eq "Administrator" -or $_.Role -eq "User") })
 
                     $restrictedGroupsCount = $restrictedGroups.Count
                     # fail the control if restricted group found on agentpool
                     if ($restrictedGroupsCount -gt 0) {
                         $controlResult.AddMessage([VerificationResult]::Failed, "Count of broader groups that have user/administrator access to agent pool: $($restrictedGroupsCount)");
-                        $formattedGroupsData = $restrictedGroups | Select @{l = 'Group'; e = { $_.Name} }, @{l = 'Role'; e = { $_.Role } }
+                        $formattedGroupsData = $restrictedGroups | Select @{l = 'Group'; e = { $_.Name} },@{l = 'Id'; e = { $_.Id } }, @{l = 'Role'; e = { $_.Role } }
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:427]: No need to display $_.Id in the logs,
Please create a separate variable which has additional data required and then assign it to $controlResult.BackupControlState

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 [hunk 2]
```diff
@@ -450,4 +454,60 @@ class AgentPool: ADOSVTBase
 
         return $controlResult;
     }
+
+    hidden [ControlResult] CheckBroaderGroupAccessAutomatedFix ([ControlResult] $controlResult) {
+        try {
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:459]: Although command gives "Fixed" status, the new permissions are not getting reflected
pls check

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1 [hunk 3]
```diff
@@ -901,11 +901,15 @@ class ServiceConnection: ADOSVTBase
                     # fail the control if restricted group found on service connection
                     if ($restrictedGroupsCount -gt 0) {
                         $controlResult.AddMessage([VerificationResult]::Failed, "Count of broader groups that have user/administrator access to service connection: $($restrictedGroupsCount)")
-                        $formattedGroupsData = $restrictedGroups | Select @{l = 'Group'; e = { $_.Name} }, @{l = 'Role'; e = { $_.Role } }
+                        $formattedGroupsData = $restrictedGroups | Select @{l = 'Group'; e = { $_.Name} },@{l = 'Id'; e = { $_.Id} }, @{l = 'Role'; e = { $_.Role } },@{l = 'AccessDisplayName'; e = { $_.AccessDisplayName } }
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1:904]: No need to display $_.Id in the logs,

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1 [hunk 4]
```diff
@@ -295,7 +295,7 @@ class VariableGroup: ADOSVTBase
                     if ($this.checkInheritedPermissionsPerVarGrp -eq $false) {
                         $responseObj = $responseObj  | where-object { $_.access -ne "inherited" }
                     }
-                    $roleAssignments += ($responseObj  | Select-Object -Property @{Name="Name"; Expression = {$_.identity.displayName}}, @{Name="Role"; Expression = {$_.role.displayName}});
+                    $roleAssignments += ($responseObj  | Select-Object -Property @{Name="Name"; Expression = {$_.identity.displayName}},@{Name="Id"; Expression = {$_.identity.id}}, @{Name="Role"; Expression = {$_.role.displayName}});
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1:298]: same

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1 [hunk 5]
```diff
@@ -327,9 +331,64 @@ class VariableGroup: ADOSVTBase
...
+                    if ($body.length -gt 1) {$body += ","}
+                    $body += @"
+                        {
+                            "userId": "$($identity.id)",
+                            "roleName": "Reader"                                                   
+                        }
+"@;
+                }
+                $RawDataObjForControlFix | Add-Member -NotePropertyName NewRole -NotePropertyValue "Reader"
+                $RawDataObjForControlFix = @($RawDataObjForControlFix  | Select-Object @{Name="DisplayName"; Expression={$_.group}}, @{Name="OldRole"; Expression={$_.Role}},@{Name="NewRole"; Expression={$_.NewRole}})
```
Comment (Garima-msft) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1:354]: In detailed logs, display name is empty. Pls check

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 (hunk 1 above)
Comment (v-himkam) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1]: done

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1 (hunk 2 above)
Comment (v-himkam) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1:460]: Fixed

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1 (hunk 3 above)
Comment (v-himkam) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.ServiceConnection.ps1]: done

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1 (hunk 5 above)
Comment (v-himkam) [reply src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.VariableGroup.ps1]: Fixed

**Review by Garima-msft — APPROVED** [review]

**Review by SaikumarRagala — APPROVED** [review]

**Review by v-soukum — APPROVED** [review]

**Pull request merged**